code/game/objects/items/stacks/stack.dm

```diff
@@ -207,10 +207,33 @@
 			to_chat(usr, "<span class='warning'>There is another [R.title] here!</span>")
 			return FALSE
 
+		if(R.one_per_turf)
+			var/list/oneperturf = list(
+				/obj/structure/girder,
+				/obj/structure/falsewall,
+				/obj/structure/window,
+				/obj/structure/cult,
+				/obj/structure/table,
+				/obj/structure/chair,
+				/obj/structure/bed,
+				/obj/machinery/computer,
+				/obj/structure/computerframe,
+				/obj/machinery/door,
+				/obj/machinery/porta_turret
+			)
+			var/obj/S = locate() in get_turf(src)
```

Comment:
So what happens if `locate()` pulls a different object on that turf first as opposed to the one you are trying to build against?

Reply:
The "locate()" looks for and makes a list of any obj in the tile in question, the other check then checks if any of the obj are on the list of obj that will not allow anything built on top of them. This means that any obj that are not found in the list will be located, but not acted on. As far as my tests go I didnt find the potential issue you refer to at least. Secondly the code already there works similarily, but checks only for one spesific type, the same type as what you are making, which is why people currently cannot make a wall on a wall, but can make a wall on a computer.

I would not that I need to change the /window to something more spesific, as it will currently not allow construction of more than one directional window per tile. So dont merge it until that is done.

Reply:
I don't think this will work at all. Doing `var/obj/S = locate()` with no args means *any* `/obj` will be returned here. Likely the first object in the `contents` of the turf (not 100% sure about that though).

Does the above code block not already accomplish what you're trying to do here?
```dm
if(R.one_per_turf && (locate(R.result_type) in get_turf(src)))
	to_chat(usr, "<span class='warning'>There is another [R.title] here!</span>")
	return FALSE
```